### PR TITLE
Updated librairy

### PR DESCRIPTION
**Pull request by Warps73**
-Added sendSms api functionnality
-Updated test
-Updated readme

### CODE REVIEW

File: src/MyElefant.php [hunk 1]
```diff
@@ -215,8 +216,58 @@ public function sendSms($campaignId, $campaignName, $contacts, $sendDate = null,
...
+            }
+        }
+        return false;
+    }
+
+    /**
+     * Sending sms
+     *
+     * @param string $campaignId Campaign's id
+     * @param array  $content
```
Comment (BasileTrujillo) [inline src/MyElefant.php]: Documentation is missing

**Review by BasileTrujillo — APPROVED** [review]

File: README.md [hunk 2]
```diff
@@ -106,6 +106,29 @@ Field's formats :
 - Sender :
 
    String
+   
+**To send sms with existing campaign**
+
+Important : To send sms with custom field, you must using tag when you creating campaign in MyElefant interface.
```
Comment (bouiboui) [inline README.md]:
```suggestion
Important : To send SMS messages with custom fields, you must use tags when you create the campaign in the MyElefant interface.
```

File: README.md [hunk 3]
```diff
@@ -106,6 +106,29 @@ Field's formats :
...
+
+Example : 
+
+[![Example](https://image.noelshack.com/fichiers/2019/08/4/1550760212-doc.png)](https://image.noelshack.com/fichiers/2019/08/4/1550760212-doc.png)
+
+then, in the code :  
+
+        <?php
```
Comment (bouiboui) [inline README.md]:
```suggestion
```php
        <?php
```

File: README.md [hunk 4]
```diff
@@ -106,6 +106,29 @@ Field's formats :
...
+
+        use myelefant\MyElefant;
+
+        $client = new MyElefant(['secretKey' => '***SECRET_KEY***'])
+
+        $client->sendSms(
+                        'campaignId',
+                         ['33612345678','This content replace [[B]]','This content replace [[C]]','This content replace [[D]]']
+                        );
+
```
Comment (bouiboui) [inline README.md:137]:
```suggestion

\```

```

File: README.md [hunk 5]
```diff
@@ -106,6 +106,29 @@ Field's formats :
...
+        <?php
+
+        use myelefant\MyElefant;
+
+        $client = new MyElefant(['secretKey' => '***SECRET_KEY***'])
+
+        $client->sendSms(
+                        'campaignId',
+                         ['33612345678','This content replace [[B]]','This content replace [[C]]','This content replace [[D]]']
```
Comment (bouiboui) [inline README.md]:
```suggestion
                         [
                            '33612345678',
                            'This content replace [[B]]',
                            'This content replace [[C]]',
                            'This content replace [[D]]'
                         ]
```

Comment (bouiboui) [inline README.md]: If 33612345678 is a real campaign ID, please replace with a placeholder

File: src/MyElefant.php [hunk 6]
```diff
@@ -170,9 +171,9 @@ private function setAuthentification($secretKey)
      * @param string|null $sender Custom sender|Template sender
      *
      * @throws @Exception
-     * @return void
+     * @return bool
```
Comment (bouiboui) [inline src/MyElefant.php:174]:
```suggestion
     * @return bool Returns true if the campaign was created succesfully
```

File: src/MyElefant.php [hunk 7]
```diff
@@ -215,8 +216,58 @@ public function sendSms($campaignId, $campaignName, $contacts, $sendDate = null,
...
+        } catch (\Throwable $e) {
+            $this->setLog('critical', $e->getMessage());
+            throw new Exception($e->getMessage());
+        }
+        if ($response->getStatusCode() == 200) {
+            $body     = $response->getBody();
+            $arr_body = json_decode($body);
+            if ($arr_body->success == true) {
+                return true;
             }
```
Comment (bouiboui) [inline src/MyElefant.php:268]: You should **always** use strict comparison operators (`===`) unless you have a very good reason not do so.
In this case, you can even simplify the return:
```php
return $arr_body->success === true;
```
or
```php
return (bool)$arr_body->success;
```

File: src/MyElefant.php [hunk 8]
```diff
@@ -325,7 +376,7 @@ private function checkFields($message = null, $sender = null)
     }
 
     /**
-     * Checking the format of the contact's array
+     * Checking the format of the contacts's array
```
Comment (bouiboui) [inline src/MyElefant.php]:
```suggestion
     * Checking the format of the contacts array
```

File: src/MyElefant.php [hunk 9]
```diff
@@ -340,4 +391,19 @@ private function checkContactsFormat($contacts)
         }
         return true;
     }
+
+    /**
+     * Checking the format of the contact's array
+     *
+     * @param array $content recipient && optional content. ex: ['33612345678','Optional content','...']
+     *
+     * @return bool
```
Comment (bouiboui) [inline src/MyElefant.php]:
```suggestion
     * @return bool Returns true if the content format is valid
```

File: src/MyElefant.php [hunk 10]
```diff
@@ -340,4 +391,19 @@ private function checkContactsFormat($contacts)
...
+     */
+    private function checkSendSmsContentFormat($content)
+    {
+        if (!is_array($content)) {
+            return false;
+        }
+        return true;
```
Comment (bouiboui) [inline src/MyElefant.php]: Can be simplified:
```php
return is_array($content);
```

File: src/tests/units/MyElefant.php [hunk 11]
```diff
@@ -172,12 +172,12 @@ function () {
...
     }
 
-    public function testSendSms()
+    public function testCreateCampaign()
     {
         $this
         ->given($MyElefantFaker = new FakeMyElefant)
             ->variable(
-                $MyElefantFaker->sendSms(
+                $MyElefantFaker->createCampaign(
```
Comment (bouiboui) [inline src/tests/units/MyElefant.php]: It should be camelCase `$myElefantFaker`

File: README.md (hunk 5 above)
Comment (Warps73) [reply README.md]: No, it's a phone number

File: README.md [hunk 12]
```diff
@@ -39,25 +39,26 @@ This ID is displayed in the list of your campaigns when you click on "Show IDs".
 
 You can send a campaign with the parameters provided when creating the campaign (message and sender) or by using custom parameters.
 
-To send a new campaign : 
+**To create and send new campaign :**
```
Comment (de-luca) [inline README.md]: You can drop the `To` in order to use the imperative

**Comment by de-luca** [comment]
As a general advice you should use the imperative for comments and when describing behavior (ie: in README)

**Pull request merged**